xtask/src/main.rs

```diff
@@ -0,0 +1,105 @@
+use clap::CommandFactory;
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+type DynError = Box<dyn std::error::Error>;
+
+fn main() {
+    if let Err(e) = try_main() {
+        eprintln!("{}", e);
+        std::process::exit(-1);
+    }
+}
+
+fn try_main() -> Result<(), DynError> {
+    let task = env::args().nth(1);
+    match task.as_ref().map(|it| it.as_str()) {
+        Some("dist") => dist()?,
+        Some("man") => dist_manpage()?,
+        _ => print_help(),
+    }
+    Ok(())
+}
+
+fn print_help() {
+    eprintln!(
+        "Tasks:
+dist            builds application and man pages
+man             build man pages
+"
+    )
+}
+
+fn dist() -> Result<(), DynError> {
+    let _ = fs::remove_dir_all(&dist_dir());
+    fs::create_dir_all(&dist_dir())?;
+
+    dist_binary()?;
+    dist_manpage()?;
+
+    Ok(())
+}
+
+fn dist_binary() -> Result<(), DynError> {
+    let cargo = env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
+    let status = Command::new(cargo)
+        .current_dir(project_root())
+        .args(&["build", "--release"])
+        .status()?;
+
+    if !status.success() {
+        Err("cargo build failed")?;
+    }
+
+    let dst = project_root().join("target/release/iroh");
+
+    fs::copy(&dst, dist_dir().join("iroh"))?;
+
+    if Command::new("strip")
+        .arg("--version")
+        .stdout(Stdio::null())
+        .status()
+        .is_ok()
+    {
+        eprintln!("stripping the binary");
+        let status = Command::new("strip").arg(&dst).status()?;
+        if !status.success() {
+            Err("strip failed")?;
+        }
+    } else {
+        eprintln!("no `strip` utility found")
+    }
+
+    Ok(())
+}
+
+fn dist_manpage() -> Result<(), DynError> {
+    let app = iroh::run::Cli::command();
+
+    // TODO(b5): This only generates the man page for the iroh command
+    // need to recursively handle all subcommands, generating man files
+    // for them as well
+
+    let man = clap_mangen::Man::new(app);
+    let mut buffer: Vec<u8> = Default::default();
+    man.render(&mut buffer)?;
```

Comment:
instead of writing to memory you can create and open the file, create a bufwriter aroubd it and the write to it